libs/blocks/locale-nav/locale-nav.js

```diff
@@ -52,9 +52,14 @@ const decorateLocales = (current) => {
   delete locales[currLocale];
   return Object.keys(locales).map((key) => {
     const prefix = key === '' ? key : `/${key}`;
-    const localePath = currLocale === ''
+    let localePath = currLocale === ''
       ? `/${key}${current.pathname}`
       : current.pathname.replace(current.locale.prefix, prefix);
+    if (localePath.startsWith('/langstore/')) {
+      localePath = localePath.replace('/langstore/', '/langstore/en/');
+      // eslint-disable-next-line no-param-reassign
+      key = 'langstore/en';
```

Comment:
Nit to avoid a bit of repetition
```suggestion
      // eslint-disable-next-line no-param-reassign
      key = 'langstore/en';
      localePath = localePath.replace('/langstore/', `/${key}/`);
```